### PR TITLE
fix(js): handle error when offset maps to space between paragraphs

### PR DESCRIPTION
**Pull request by gebrial**
CYP-1002

### CODE REVIEW

File: js/src/degausser.js [hunk 1]
```diff
@@ -43,7 +43,12 @@ export const getRangeFromOffset = (start, end, doc = document, map = null) => {
           }
         }
 
-        range.setStart(entry.node, adjustedStart + skips)
+        if (adjustedStart + skips - entry.node.length === 1){
```
Comment (pashoo2) [inline js/src/degausser.js:46]: can we add a test for that case?

Comment (gebrial) [reply js/src/degausser.js:46]: done

**Comment by kyles-ep** [comment]
Are we changing expected logic, or is this a bug and we are fixing the code to match expected logic?

**Review by kyles-ep — APPROVED** [review]

**Comment by gebrial** [comment]
> Are we changing expected logic, or is this a bug and we are fixing the code to match expected logic?

I'm certain it's a bug. What would normally happen is that it would just throw an error on that line in cases where the offset mapped to be between elements (e.g., <p> elements). This seems to be because the degausser adds a space character between elements but treats it as part of that element. This causes the range out of bounds error.

**Review by pashoo2 — APPROVED** [review]

**Pull request merged**